Convex_hull_3/test/Convex_hull_3/test_extreme_points.cpp

```diff
@@ -0,0 +1,216 @@
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/convex_hull_3.h>
+#include <CGAL/Convex_hull_traits_3.h>
+#include <CGAL/Exact_rational.h>
+#include <CGAL/Cartesian.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <vector>
+#include <cassert>
+#include <algorithm>
+#include <fstream>
+
+typedef CGAL::Exact_rational                          NT;
+typedef CGAL::Cartesian<NT>                           K;
+typedef CGAL::Convex_hull_traits_3<K>                 Traits;
+typedef Traits::Polygon_mesh                          Polyhedron_3;
+typedef K::Point_3                                    Point_3;
+
+
+void test_function_overload()
+{
+  std::vector<Point_3> points;
+  points.push_back(Point_3(0,0,0));
+  points.push_back(Point_3(10,0,0));
+  points.push_back(Point_3(0,10,0));
+  points.push_back(Point_3(0,0,10));
+  points.push_back(Point_3(5,5,5));
+  points.push_back(Point_3(2,5,3));
+  points.push_back(Point_3(1,3,2));
+
+  std::vector<Point_3> extreme_points;
+  CGAL::extreme_points_3(points, std::back_inserter(extreme_points));
+  CGAL_assertion(extreme_points.size() == 5);
+
+  Polyhedron_3 polyhedron;
+  CGAL::convex_hull_3(points.begin(), points.end(), polyhedron);
+  typedef Polyhedron_3::Point_iterator Point_iterator;
+  for (Point_iterator p_it = polyhedron.points_begin(); p_it != polyhedron.points_end(); ++p_it)
+  {
+    Point_3 p = *p_it;
+    CGAL_assertion(std::find(extreme_points.begin(), extreme_points.end(), p) != extreme_points.end());
+  }
+}
+
+void test_triangulated_cube(const char* fname)
+{
+  typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+  typedef K::Point_3 Point_3;
+  typedef CGAL::Surface_mesh<Point_3> SurfaceMesh;
+
+  std::ifstream input(fname);
+  SurfaceMesh mesh;
+  if (!input || !(input >> mesh) || mesh.is_empty()) {
+    std::cerr << fname << " is not a valid off file.\n";
+    exit(1);
+  }
+
+  std::vector<Point_3> mesh_points;
+  typedef typename boost::property_map<SurfaceMesh, boost::vertex_point_t>::type Pmap;
+  Pmap vpmap = get_property_map(boost::vertex_point, mesh);
+
+  typedef typename boost::graph_traits<SurfaceMesh>::vertex_descriptor vertex_descriptor;
+  BOOST_FOREACH(vertex_descriptor v, vertices(mesh))
+  {
+    Point_3 p = get(vpmap, v);
+    mesh_points.push_back(p);
+  }
+
+  std::vector<Point_3> extreme_points;
+  CGAL::extreme_points_3(mesh_points, std::back_inserter(extreme_points)) ;
+  CGAL_assertion(extreme_points.size() == 8);
```

Comment:
This seems to be luck, I've tried with only the points and got the same issue as with `cross.off`, not strictly convex.